src/iop/zonesystem.c

```diff
@@ -203,7 +203,7 @@ static void process_common_cleanup(struct dt_iop_module_t *self, dt_dev_pixelpip
   const int ch = piece->colors;
   const int size = d->params.size;
 
-  if(piece->pipe->mask_display) dt_iop_alpha_copy(ivoid, ovoid, width, height);
+  if(piece->pipe->mask_display & DT_DEV_PIXELPIPE_DISPLAY_MASK) dt_iop_alpha_copy(ivoid, ovoid, width, height);
```

Comment:
Yes, i understand all those reasons, but, in opencl kernels case i don't remember seeing `dt_iop_alpha_copy()` calls in the `process_cl()`, so it *seems* to be always done in the kernel.
But `dt_iop_alpha_copy()` is pretty much always the case for cpu codepath...